src/ray/gcs/gcs_server/node_info_handler_impl.cc

```diff
@@ -44,6 +55,7 @@ void DefaultNodeInfoHandler::HandleUnregisterNode(
     rpc::SendReplyCallback send_reply_callback) {
   ClientID node_id = ClientID::FromBinary(request.node_id());
   RAY_LOG(DEBUG) << "Unregistering node info, node id = " << node_id;
+  gcs_node_manager_.OnNodeFailure(node_id);
```

Comment:
> maybe rename `OnNodeFailure` to `OnNodeDisconncted`, because this is the intentional disconnection case here.

 I think it's more appropriate to use "removenode" instead.